src/DataShare/DataShare/SynchronizationSettings/NewAzDataShareSynchronizationSetting.cs

```diff
@@ -0,0 +1,129 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.DataShare.SynchronizationSetting
+{
+    using System;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.DataShare.Common;
+    using Microsoft.Azure.Commands.DataShare.Helpers;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.DataShare;
+    using Microsoft.Azure.Management.DataShare.Models;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Extensions;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Models;
+
+    /// <summary>
+    /// Defines the New-AzDataShareSynchronizationSetting cmdlet.
+    /// </summary>
+    [Cmdlet(
+         "New",
+         ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DataShareSynchronizationSetting",
+         DefaultParameterSetName = ParameterSetNames.FieldsParameterSet,
+         SupportsShouldProcess = true),
+     OutputType(typeof(PSDataShareSynchronizationSetting))]
+    public class NewAzDataShareSynchronizationSetting : AzureDataShareCmdletBase
+    {
+        /// <summary>
+        /// The resource group name of azure data share account.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
```

Comment:
As discussed above, do not specify `ValudFromPipelineByPropertyName` for an parameter other than `ResourceId`